### PR TITLE
Introduced a simple demo of mixed index and text2gremlin demonstration.

### PR DESCRIPTION
**Pull request by vichayturen**
Main changes:
1. Modified the configuration method.
2. Added vector index for RAG, supported mixed-retrieving and reranking.
3. Introduced embedding configuration.
4. Introduced index operation and document operation.
5. Added task of transferring natural language to gremlin and its demonstration.
6. Simplified the use and modify the usage in README.

---

TODOs (add them later) 🔢 
- [ ] fix github security alert
- [ ] connection check (GraphDB & LLM Server)
   - [x] openai check 
   - [ ] qianfan/wenxin check (use auth-api?)
   - [ ] local llm check 
- [x] graph error handle
	- [x] print request & response url + body in py-client
	- [x] wrapper some exception (don't throw Error detail directly)
- [x] radio (Selection)
- [x] configs improvement (remove duplicate items)
- [x] support uploading file in UI (text type)
- [x] allow users to modify prompt template (BASIC)
- [x] add a one-click graph clean button? (use [clear](https://hugegraph.apache.org/docs/clients/restful-api/graphs/#method--url-2) api)
... (check the latest [doc](https://hugegraph.feishu.cn/wiki/Es2SwB2Zoi91UUkPj6vcu1vPnAF))

### CODE REVIEW

**Comment by imbajin** [comment]
Now we could use `log.xx(msg)` to print the log: ↓
image

**Comment by ChenZiHong-Gavin** [comment]
Now the .config file indtead json file can be generated with command like `python -m hugegraph_llm.config.generate --dir_path=./`

File: hugegraph-llm/src/hugegraph_llm/config/config.py [hunk 1]
```diff
@@ -20,8 +20,11 @@
 
 from dataclasses import dataclass
 from typing import Literal, Optional
-from pathlib import Path
-from dotenv import dotenv_values
+from dotenv import dotenv_values, set_key
+
+dirname = os.path.dirname
+package_path = dirname(dirname(dirname(dirname(os.path.abspath(__file__)))))
+env_path = os.path.join(package_path, ".env")
```
Comment (jasinliu) [inline hugegraph-llm/src/hugegraph_llm/config/config.py:27]: 在pip安装之后 运行代码会在python所在路径下生成.env

Comment (ChenZiHong-Gavin) [reply hugegraph-llm/src/hugegraph_llm/config/config.py:29]: There are some aspects that can be optimized in the future.
- [ ] Add some configuration-related comments in the .env file
- [ ] Rename `.env` filename to something related to hugegraph-llm
- [ ] Unify the variables in the .env file and the variables in the Config class to be all uppercase
- [ ] Differentiate between commonly used parameters and parameters that are secret keys
- [ ] Give user a choice where to place their  .env file

**Review by ChenZiHong-Gavin — APPROVED** [review]

**Review by simon824 — APPROVED** [review]

**Review by javeme — COMMENTED** [review]
prefer to add an assertEqual for each test(like test_vector_index)

**Pull request merged**